diff --git a/node_modules/expo-modules-core/android/.project b/node_modules/expo-modules-core/android/.project
new file mode 100644
index 0000000..ea2be6b
--- /dev/null
+++ b/node_modules/expo-modules-core/android/.project
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<projectDescription>
+	<name>expo-modules-core</name>
+	<comment>Project expo-modules-core created by Buildship.</comment>
+	<projects>
+	</projects>
+	<buildSpec>
+		<buildCommand>
+			<name>org.eclipse.buildship.core.gradleprojectbuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+	</buildSpec>
+	<natures>
+		<nature>org.eclipse.buildship.core.gradleprojectnature</nature>
+	</natures>
+	<filteredResources>
+		<filter>
+			<id>1770017723704</id>
+			<name></name>
+			<type>30</type>
+			<matcher>
+				<id>org.eclipse.core.resources.regexFilterMatcher</id>
+				<arguments>node_modules|\.git|__CREATED_BY_JAVA_LANGUAGE_SERVER__</arguments>
+			</matcher>
+		</filter>
+	</filteredResources>
+</projectDescription>
diff --git a/node_modules/expo-modules-core/android/ExpoModulesCorePlugin.gradle b/node_modules/expo-modules-core/android/ExpoModulesCorePlugin.gradle
index bcf9b00..a9c049f 100644
--- a/node_modules/expo-modules-core/android/ExpoModulesCorePlugin.gradle
+++ b/node_modules/expo-modules-core/android/ExpoModulesCorePlugin.gradle
@@ -79,15 +79,18 @@ ext.useExpoPublishing = {
   }
 
   project.afterEvaluate {
-    publishing {
-      publications {
-        release(MavenPublication) {
-          from components.release
+    def releaseComponent = components.findByName("release")
+    if (releaseComponent != null) {
+      publishing {
+        publications {
+          release(MavenPublication) {
+            from releaseComponent
+          }
         }
-      }
-      repositories {
-        maven {
-          url = mavenLocal().url
+        repositories {
+          maven {
+            url = mavenLocal().url
+          }
         }
       }
     }
diff --git a/node_modules/expo-modules-core/expo-module-gradle-plugin/.project b/node_modules/expo-modules-core/expo-module-gradle-plugin/.project
new file mode 100644
index 0000000..7fed290
--- /dev/null
+++ b/node_modules/expo-modules-core/expo-module-gradle-plugin/.project
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<projectDescription>
+	<name>expo-module-gradle-plugin</name>
+	<comment>Project expo-module-gradle-plugin created by Buildship.</comment>
+	<projects>
+	</projects>
+	<buildSpec>
+		<buildCommand>
+			<name>org.eclipse.buildship.core.gradleprojectbuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+	</buildSpec>
+	<natures>
+		<nature>org.eclipse.buildship.core.gradleprojectnature</nature>
+	</natures>
+	<filteredResources>
+		<filter>
+			<id>1770017723696</id>
+			<name></name>
+			<type>30</type>
+			<matcher>
+				<id>org.eclipse.core.resources.regexFilterMatcher</id>
+				<arguments>node_modules|\.git|__CREATED_BY_JAVA_LANGUAGE_SERVER__</arguments>
+			</matcher>
+		</filter>
+	</filteredResources>
+</projectDescription>
